### PR TITLE
Add write(3) to PlatformUtility

### PR DESCRIPTION
**Pull request by hughbe**
Also run clang-format on PlatformUtility.cpp

### CODE REVIEW

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
@swift-ci please smoke test OS X Platform

File: lib/Basic/PlatformUtility.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@
 using namespace llbuild;
 using namespace llbuild::basic;
 
-bool sys::chdir(const char* fileName) {
+bool sys::chdir(const char *fileName) {
```
Comment (ddunbar) [inline lib/Basic/PlatformUtility.cpp]: I follow the Google style for placement of `*`, so the previous version was correct.

Comment (hughbe) [inline lib/Basic/PlatformUtility.cpp]: Oops, can revert if you'd like?

**Pull request merged**